internal/eru/agent/metrics.go

```diff
@@ -0,0 +1,77 @@
+package agent
+
+import (
+	"context"
+	"sync/atomic"
+
+	"github.com/patrickmn/go-cache"
+	"github.com/projecteru2/core/log"
+	"github.com/projecteru2/core/utils"
+	virttypes "github.com/projecteru2/libyavirt/types"
+	"github.com/projecteru2/yavirt/configs"
+	"github.com/projecteru2/yavirt/internal/eru/types"
+	"github.com/projecteru2/yavirt/internal/vmcache"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	vmHealthyDesc = prometheus.NewDesc(
+		prometheus.BuildFQName("vm", "", "healthy"),
+		"VM healthy status.",
+		[]string{"EruID", "node", "app_id", "app_sid", "appname", "ip"},
+		nil)
+	coreHealthyDesc = prometheus.NewDesc(
+		prometheus.BuildFQName("node", "core", "healthy"),
+		"core healthy status.",
+		[]string{"node"},
+		nil)
+)
+
+type MetricsCollector struct {
+	wrkStatusCache *cache.Cache
+	coreHealthy    atomic.Bool
+}
+
+func (mgr *Manager) GetMetricsCollector() *MetricsCollector {
+	return mgr.mCol
+}
+
+func (e *MetricsCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- vmHealthyDesc
+}
+
+func (e *MetricsCollector) Collect(ch chan<- prometheus.Metric) {
+	for _, v := range e.wrkStatusCache.Items() {
+		wrkStatus, _ := v.Object.(*types.WorkloadStatus)
+		if !wrkStatus.Running {
```

Comment:
prefer to check `wrkStatus` if is nil in the first

Reply:
I'll add a check here, actually wrkStatus never be nil here